docs/design/StratisSoftwareDesign.lyx

```diff
@@ -6933,5 +6933,423 @@ Based on looking at the existing available building blocks, the best option
+\end_layout
+
+\begin_layout Subsection
+Encrypted device activation
+\end_layout
+
+\begin_layout Standard
+Devices are activated with the activation name in the Stratis token.
+ See 
+\begin_inset CommandInset ref
+LatexCommand ref
+reference "subsec:Stratis-token"
+plural "false"
+caps "false"
+noprefix "false"
+
+\end_inset
+
+ for an explanation of the Stratis token.
+ libcryptsetup handles device activation and should result device with a
+ path of 
+\begin_inset listings
+inline false
+status collapsed
+
+\begin_layout Plain Layout
+
+/dev/mapper/<activation_name>
+\end_layout
+
+\end_inset
+
+ where the unencrypted data can be accessed.
+\end_layout
+
+\begin_layout Subsection
+Encrypted device destruction
+\end_layout
+
+\begin_layout Standard
+Destroying an encrypted Stratis device does not wipe the entire device due
+ to the time that this would take increasing linearly with regard to disk
+ size.
+ Instead, Stratis uses libcryptsetup to destroy all of the keyslots, wiping
+ all of the encrypted MEK (media encryption key) data in each keyslot.
+ It then obtains the size of the LUKS2 metadata on this device and does
+ an additional wipe of the remaining LUKS2 metadata.
+ This provides the following device properties on pool destruction:
+\end_layout
+
+\begin_layout Itemize
+The MEK data is destroyed so the data stored on the device will be unrecoverable.
+\end_layout
+
+\begin_layout Itemize
+If this device is used when creating another pool later, Stratis will be
+ able to reinitialize it and will consider this device unowned due to the
+ wipe of the LUKS2 metadata.
+\end_layout
+
+\begin_layout Itemize
+The encrypted data will be left on the device but will be inaccessible due
+ to the destruction of the key.
+\end_layout
+
+\begin_layout Standard
+If the user is concerned about leaving the encrypted data on the device
+ even if the key has been destroyed, the disk should be zeroed or wiped
+ with random data.
+\end_layout
+
+\begin_layout Subsection
+Key consistency for encrypted pools
+\end_layout
+
+\begin_layout Standard
+Stratis takes steps to ensure that the same passphrase used during initial
+ creation of the pool is the passphrase that is used for adding encrypted
+ block devices at a later time.
+ stratisd runs a check prior to adding block devices to an encrypted pool
+ to verify that the passphrase in the kernel keyring with the key description
+ recorded in the LUKS2 metadata can unlock the block devices that have already
+ been encrypted and added to the pool.
+ This prevents the passphrase data associated with a key description from
+ being accidentally or intentionally changed between pool creation and addition
+ of encrypted block devices which lead to a broken pool on a later activation
+ of the pool.
+\end_layout
+
+\begin_layout Subsection
+Key management
+\end_layout
+
+\begin_layout Standard
+Currently Stratis requires the user to manage the kernel keyring.
```

Comment:
You can rewrite this entirely, starting out by saying something like:

```
Stratis provides facilities for key management.
```

Then just describe what they are and what makes them secure.